erpnext/hr/doctype/department_approver/department_approver.py

```diff
@@ -26,7 +26,12 @@ def get_approvers(doctype, txt, searchfield, start, page_len, filters):
 		department_list = frappe.db.sql("""select name from `tabDepartment` where lft <= %s
 			and rgt >= %s
 			and disabled=0
-			order by lft desc""", (department_details.lft, department_details.rgt), as_list = True)
+			order by lft desc""", (department_details.lft, department_details.rgt), as_list=True)
+
+	leave_approver = frappe.get_cached_value('Employee', filters.get('employee'), 'leave_approver')
+	if leave_approver:
+		approver = frappe.db.get_value("User", leave_approver, ['name', 'first_name', 'last_name'])
+		approvers.append(approver)
```

Comment:
If leave approver defined on employee, then only return that approver. Otherwise, fetch from department

Reply:
added change